CONTRIBUTING.md

```diff
@@ -48,6 +48,7 @@ CSharp: # required, this will be the name of the enum variant for the language a
       - "#803788"
       - "#FFFFFF"
     chip: "#178600" # required, this is used for the language breakdown bar, its value can be found in linguist (link 2).
+  icon: "\uE648" # optional, the UTF-16 code point of the nerd font icon if supported (link 3).
```

Comment:
AFAIK `\u1234` will only work sometimes, while `\U12345678` will always work, so you might want to recommend that to avoid confusion.

Reply:
Good point, I missed that part 😓 

Following on https://github.com/o2sh/onefetch/issues/1308#issuecomment-2293861246, here is my suggestion :

1. In `languages.yaml` use the nerd font code point verbatim without the "\u" which is sequence of 4 to 5 (maybe 6) hex digits. 
2. In `language.tera` do the transformation to comply with Rust syntax on unicode code points : hex_digit+ 6 -> '\u' '{' hex_digit+ 6 '}'

the method would look smth like:

```
    pub fn get_icon(&self) -> Option<&'static str> {
        match self {
            {% for language, attrs in languages -%}
                {% if attrs.icon is defined %}Language::{{ language }} => Some("\u{{ '{' }}{{ attrs.icon }}{{ '}' }}"),{% else %}Language::{{ language }} => None,{% endif %}
            {% endfor %}
        }
    }
```

As a result the generated rust code would look smth like this:

```
Language::Emojicode => Some("\u{f1044}"),
Language::Erlang => Some("\u{E7B1}"),
Language::FSharp => Some("\u{E7A7}"),
Language::Fish => Some("\u{EE41}"),
Language::Forth => Some("\u{229E}"),
Language::FortranLegacy => Some("\u{f121a}"),
Language::FortranModern => Some("\u{f121a}"),
```

Reply:
for that we would need to use single quotes :

```
  icon: '\u{E6B5}'

```

Reply:
[8a72a45](https://github.com/o2sh/onefetch/pull/1395/commits/8a72a45f594cc027f28231a00332133148d0e5fd)

Reply:
@o2sh 
Should `pub fn get_icon(&self)` return `Option<char>` instead of `Option<&'static str>`?
It wasn't this way originally because the encoded char threw an error about multiple codepoints, however this doesn't seem to be the case with the escapes.

```rust
    pub fn get_icon(&self) -> Option<char> {
        match self {
            {% for language, attrs in languages -%}
                {% if attrs.icon is defined %}Language::{{ language }} => Some('{{ attrs.icon }}'),{% else %}Language::{{ language }} => None,{% endif %}
            {% endfor %}
        }
    }
```

Reply:
Good suggestion!
Thanks for pointing it out